sklearn/ensemble/stacking_model.py

```diff
@@ -0,0 +1,173 @@
+# libraries
+import numpy as np
+
+# scikit-learn base libraries
+from sklearn.base import BaseEstimator, ClassifierMixin, RegressorMixin
+
+# scikit-learn modules
+from sklearn.cross_validation import cross_val_predict
+from sklearn.ensemble import VotingClassifier
+
+
+class StackingClassifier(BaseEstimator,ClassifierMixin):
+    '''
+    stacking ensemble classifier based on scikit-learn
+    '''
+    def __init__(self,stage_one_clfs,stage_two_clfs,weights=None, n_runs=10):
+        '''
+        
+        weights: weights of the stage_two_clfs
+        n_runs: train stage_two_clfs n_runs times and average them (only for probabilistic output)
+        '''
+        self.stage_one_clfs = stage_one_clfs
+        self.stage_two_clfs = stage_two_clfs
+        if weights == None:
+            self.weights = [1] * len(stage_two_clfs)
+        else:
+            self.weights = weights
+        self.n_runs = n_runs
+        
+    def fit(self,X,y):
+        '''
+        fit the model
+        '''
+        self.__X = X
+        self.__y = y
+        
+        # fit the first stage models
+        for clf in self.stage_one_clfs:
+            y_pred = cross_val_predict(clf, X, y, cv=5, n_jobs=1)

```

Comment:
Actually, I'm not clear if we have to use `cross_val_x` here. The training data since you have done `clf.fit(X, y)` below and the data from which you obtain predictions are no more independent are they?

Again reference: http://machine-learning.martinsewell.com/ensembles/stacking/

Maybe @jnothman has some better idea on how to handle this in terms of API?